pkg/engine/mutate/strategicMergePatch_test.go

```diff
@@ -25,33 +24,149 @@ func TestMergePatch(t *testing.T) {
 		},
 		{
 			// condition matches the first element of the array
-			rawPolicy:   []byte(`{"spec": {"containers": [{"(image)": "gcr.io/google-containers/busybox:*"}],"imagePullSecrets": [{"name": "regcred"}]}}`),
-			rawResource: []byte(`{"apiVersion": "v1","kind": "Pod","metadata": {"name": "hello"},"spec": {"containers": [{"name": "hello","image": "gcr.io/google-containers/busybox:latest"},{"name": "hello2","image": "gcr.io/google-containers/busybox:latest"},{"name": "hello3","image": "gcr.io/google-containers/nginx:latest"}]}}`),
-			expected:    []byte(`{"apiVersion":"v1","kind":"Pod","metadata":{"name":"hello"},"spec":{"containers":[{"image":"gcr.io/google-containers/busybox:latest","name":"hello"},{"image":"gcr.io/google-containers/busybox:latest","name":"hello2"},{"image":"gcr.io/google-containers/nginx:latest","name":"hello3"}],"imagePullSecrets":[{"name":"regcred"}]}}`),
+			rawPolicy: []byte(`{
+        "spec": {
+          "containers": [
+            {
+              "(image)": "gcr.io/google-containers/busybox:*"
+            }
+          ],
+          "imagePullSecrets": [
+            {
+              "name": "regcred"
+            }
+          ]
+        }
+      }`),
+			rawResource: []byte(`{
+        "apiVersion": "v1",
+        "kind": "Pod",
+        "metadata": {
+          "name": "hello"
+        },
+        "spec": {
+          "containers": [
+            {
+              "name": "hello",
+              "image": "gcr.io/google-containers/busybox:latest"
+            },
+            {
+              "name": "hello2",
+              "image": "gcr.io/google-containers/busybox:latest"
+            },
+            {
+              "name": "hello3",
+              "image": "gcr.io/google-containers/nginx:latest"
+            }
+          ]
+        }
+      }`),
+			expected: []byte(`{
+        "apiVersion": "v1",
+        "kind": "Pod",
+        "metadata": {
+          "name": "hello"
+        },
+        "spec": {
+          "containers": [
+            {
+              "image": "gcr.io/google-containers/busybox:latest",
+              "name": "hello"
+            },
+            {
+              "image": "gcr.io/google-containers/busybox:latest",
+              "name": "hello2"
+            },
+            {
+              "image": "gcr.io/google-containers/nginx:latest",
+              "name": "hello3"
+            }
+          ],
+          "imagePullSecrets": [
+            {
+              "name": "regcred"
+            }
+          ]
+        }
```

Comment:
Isn't the conditional anchor works only on its child tags? Why the `imagePullSecrets` is added to the Pod? Same for the next unit test.

Reply:
Because `imagePullSecrets` is not a child tag for `image`.